src/backend/routers/routes.test.ts

```diff
@@ -0,0 +1,61 @@
+import test from "ava";
+import { trpcRouter } from "@/backend/routers/_app";
+import { createContext } from "@/backend/context";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { TRPCError } from "@trpc/server";
+
+test("All API endpoints are auth guarded", async (t) => {
+  // Create a more complete mock of req and res
+  const mockReq = {
+    headers: {},
+    cookies: {},
+  } as unknown as NextApiRequest;
+
+  const mockRes = {
+    getHeader: () => undefined,
+    setCookie: () => void 0,
+    setHeader: () => void 0,
```

Comment:
I'm curious, what's the reason that these are defined as returning void 0 instead of undefined?

Reply:
@KCCPMG typescript was complaining ... some esoteric type safety problem, I forget the details but initially it was returning undefined for all here.

Reply:
Hm ok. I changed it to undefined on my end when I was playing around with it and didn't get any type issues, maybe differences in our TS extensions?

Reply:
you're totally right, changed to undefined ... I was having Cursor's AI fix linting/etc problems, it must have went overboard.